jung-samples/src/main/java/edu/uci/ics/jung/samples/SubLayoutDemo.java

```diff
@@ -333,24 +329,18 @@ private void cluster(boolean state) {
 
           LayoutAlgorithm<String, Point2D> subLayoutAlgorithm =
               getLayoutAlgorithmFor(subLayoutType);
-          //          subLayoutAlgorithm.setInitializer(new RandomLocationTransformer<String>(subLayoutSize));
-          //          subLayout.setSize(subLayoutSize);
-          //          clusteringLayoutModel.put(subLayoutAlgorithm, center);
+
           LayoutModel<String, Point2D> newLayoutModel =
-              new LoadingCacheLayoutModel<>(
-                  subGraph.asGraph(),
-                  POINT_MODEL,
-                  subLayoutSize.width,
-                  subLayoutSize.height,
-                  0,
-                  new RandomLocationTransformer<>(
-                      POINT_MODEL, subLayoutSize.width, subLayoutSize.height, 0));
-
-          //          VisualizationModel newModel =
-          //              new BaseVisualizationModel(subGraph, subLayoutAlgorithm, subLayoutSize);
-          //          newModel.getLayout().setInitializer(new RandomLocationTransformer<String,Point2D>(subLayoutSize));
+              LoadingCacheLayoutModel.<String, Point2D>builder()
+                  .setGraph(subGraph.asGraph())
+                  .setPointModel(POINT_MODEL)
+                  .setSize(subLayoutSize.width, subLayoutSize.height)
+                  .setInitializer(
```

Comment:
A few things about this:

(1) I think that it would make sense for the RandomLocationTransformer to be the default initializer.  
That way it doesn't have to be specified in most cases.

(2) If the point model and dimensions of the initializer must be the same as those provided to the `LayoutModel` proper, then specifying it twice is both tedious and error-prone.  Perhaps we should just be having the user specify (via an enum? not sure...) what kind of initializer to use.

(Side point about `RandomLocationTransformer`: that should basically be the initializer of last resort; ideally we would use knowledge of the topology of the graph to inform the decision of where to put the starting positions of the nodes.)

Reply:
Easy follow-up PR for me or someone.